src/server/generic_family.cc

```diff
@@ -1463,29 +1487,57 @@ void GenericFamily::Sort(CmdArgList args, const CommandContext& cmd_cntx) {
       }
       bounds = {offset, limit};
       i += 2;
+    } else if (arg == "STORE") {
+      if (i + 1 >= args.size()) {
+        return builder->SendError(kSyntaxErr);
+      }
+      store_key = ArgS(args, i + 1);
+      i += 1;
     } else {
       LOG_EVERY_T(ERROR, 1) << "Unsupported option " << arg;
       return builder->SendError(kSyntaxErr, kSyntaxErrType);
     }
   }
 
-  OpResultTyped<SortEntryList> fetch_result =
-      cmd_cntx.tx->ScheduleSingleHopT([&](Transaction* t, EngineShard* shard) {
-        return OpFetchSortEntries(t->GetOpArgs(shard), key, alpha);
-      });
+  ShardId source_sid = Shard(key, shard_set->size());
+  OpResultTyped<SortEntryList> fetch_result;
+  auto fetch_cb = [&](Transaction* t, EngineShard* shard) {
+    ShardId shard_id = shard->shard_id();
+    if (shard_id == source_sid) {
+      fetch_result = OpFetchSortEntries(t->GetOpArgs(shard), key, alpha);
+    }
+    return OpStatus::OK;
+  };
+
+  if (store_key) {
+    cmd_cntx.tx->Execute(std::move(fetch_cb), false);
+  } else {
+    cmd_cntx.tx->Execute(std::move(fetch_cb), true);
+  }
+
+  // OpResultTyped<SortEntryList> fetch_result =
+  //     cmd_cntx.tx->ScheduleSingleHopT([&](Transaction* t, EngineShard* shard) {
+  //       return OpFetchSortEntries(t->GetOpArgs(shard), key, alpha);
+  //     });
 
-  if (fetch_result == OpStatus::WRONG_TYPE)
+  if (fetch_result == OpStatus::WRONG_TYPE) {
+    cmd_cntx.tx->Conclude();
     return builder->SendError(fetch_result.status());
+  }
 
-  if (fetch_result.status() == OpStatus::INVALID_NUMERIC_RESULT)
+  if (fetch_result.status() == OpStatus::INVALID_NUMERIC_RESULT) {
+    cmd_cntx.tx->Conclude();
     return builder->SendError("One or more scores can't be converted into double");
+  }
 
   auto* rb = static_cast<RedisReplyBuilder*>(builder);
-  if (!fetch_result.ok())
+  if (!fetch_result.ok()) {
```

Comment:
I would unify all the non-ok conditions here:
```
if (!fetch_result.ok()) {
   cmd_cntx.tx->Conclude();
   if (fetch_result.status() == OpStatus::INVALID_NUMERIC_RESULT) 
     return builder->SendError(...);
   ....
}
```

Reply:
yup that will be better. Will do that